### PR TITLE
Fixed unnecessary entries in table `api-session` when using insta-login in API calls

### PR DESCRIPTION
**Pull request by kiatng**
…in in API calls.

### Description (*)
The API session has default expiry of 3600 secs from the time of the last API call. This expiry is managed in the table `api_asession`.  With the introduction of insta-login in PR #3443, the session expiry become irrelevant. This PR bypassed adding entries to the various api tables. In my test, this saves 10 to 20 mini-seconds to insta-login.

### Related Pull Requests
PR #3443 

### Fixed Issues (if relevant)
Partially fixed issue #3449.

### Manual testing scenarios (*)
1. No new entries are added in the table `api_asession` with insta-login.
2. Normal entries are added in the table `api_asession` with "normal" login.

### CODE REVIEW

File: app/code/core/Mage/Api/Model/Server/Handler/Abstract.php [hunk 1]
```diff
@@ -100,16 +115,6 @@ protected function _isAllowed($resource, $privilege = null)
...
     }
 
-    /**
-     *  Check session expiration
-     *
-     *  @return  bool
-     */
-    protected function _isSessionExpired()
-    {
-        return $this->_getSession()->isSessionExpired();
```
Comment (kiatng) [inline app/code/core/Mage/Api/Model/Server/Handler/Abstract.php:110]: Note: This line has a bug, `isSessionExpired()` expects a param `$user`:
https://github.com/OpenMage/magento-lts/blob/2a2a2fb504247e8966f8ffc2e17d614be5d43128/app/code/core/Mage/Api/Model/Session.php#L186-L186

I've checked, this function is not used.

**Review by colinmollenhour — APPROVED** [review]
Very nice!

File: app/code/core/Mage/Api/Model/Session.php [hunk 2]
```diff
@@ -105,8 +127,15 @@ public function clear()
     public function login($username, $apiKey)
     {
         $user = Mage::getModel('api/user')
-            ->setSessid($this->getSessionId())
-            ->login($username, $apiKey);
+            ->setSessid($this->getSessionId());
+        if ($this->getIsInstaLogin() && $user->authenticate($username, $apiKey)) {
+            Mage::dispatchEvent('api_user_authenticated', [
```
Comment (fballiano) [inline app/code/core/Mage/Api/Model/Session.php:132]: do we need to dd this event to the documentation?

Comment (kiatng) [reply app/code/core/Mage/Api/Model/Session.php:132]: No need to document, this event is carried from here:
https://github.com/OpenMage/magento-lts/blob/2a2a2fb504247e8966f8ffc2e17d614be5d43128/app/code/core/Mage/Api/Model/User.php#L263-L275

It is dispatched after authentication, added here for consistency.

**Comment by fballiano** [comment]
we can remove a line from the phpstan baseline in order to fix the workflow

**Comment by fballiano** [comment]
it says that we've to remove that line from the baseline, since it's not triggering an error anymore

**Review by elidrissidev — APPROVED** [review]

**Review by fballiano — APPROVED** [review]

**Pull request merged**